tf2_gnn/models/graph_regression_task.py

```diff
@@ -26,46 +34,121 @@ def __init__(self, params: Dict[str, Any], dataset: GraphDataset, name: str = No
         super().__init__(params, dataset=dataset, name=name)
         self._node_to_graph_aggregation = None
 
+        # Construct sublayers:
+        self._weighted_avg_of_nodes_to_graph_repr = WeightedSumGraphRepresentation(
+            graph_representation_size=self._params["graph_representation_size"] // 2,
+            num_heads=self._params["graph_aggregation_num_heads"] // 2,
+            weighting_fun="softmax",
+            scoring_mlp_layers=[
+                l // 2 for l in self._params["graph_aggregation_layers"]
+            ],
+            scoring_mlp_dropout_rate=self._params["graph_aggregation_dropout_rate"],
+            scoring_mlp_activation_fun="elu",
+            transformation_mlp_layers=[
+                l // 2 for l in self._params["graph_aggregation_layers"]
+            ],
+            transformation_mlp_dropout_rate=self._params[
+                "graph_aggregation_dropout_rate"
+            ],
+            transformation_mlp_activation_fun="elu",
+        )
```

Comment:
Shouldn't we first assert that `graph_representation_size` is even?

Also, I see we're halving some other things, such as MLP hidden layer sizes. Are we assuming the user set them having the full graph representation size in mind, and we're scaling them to be appropriate for a half?

Reply:
Yeah, this is a bit ugly. Primarily, I copied this from another place where I wanted to keep backwards compat in sizes, but maybe that's the wrong approach here. I'll clean this up.